### ISSUE TITLE
sed: can't read /opt/bitnami/mongodb/conf/mongodb.conf: No such file or directory

### ISSUE THREAD

**Issue by Jaysins** [bitnami/charts#24387]
### Name and Version

bitnami/mongo:7.0.6

### What architecture are you using?

None

### What steps will reproduce the bug?

helm show valuesbitnamicharts/mongodb -  > mongovalues.yml

change architecture to replicaset

`helm install mongo -f mongovalues.yml oci://registry-1.docker.io/bitnamicharts/mongodb` 

### What is the expected behavior?

install mongo replicaset

### What do you see instead?
```console
$ kubectl logs -f mongo-mongodb-0
mongodb 02:54:12.06 INFO  ==> Advertised Hostname: mongo-mongodb-0.mongo-mongodb-headless.mongo-staging-6.svc.cluster.local
mongodb 02:54:12.06 INFO  ==> Advertised Port: 27017
mongodb 02:54:12.06 INFO  ==> Data dir empty, checking if the replica set already exists
MongoNetworkError: connect ECONNREFUSED 10.1.0.149:27017
mongodb 02:54:13.34 INFO  ==> Pod name matches initial primary pod name, configuring node as a primary
mongodb 02:54:13.35 
mongodb 02:54:13.35 Welcome to the Bitnami mongodb container
mongodb 02:54:13.35 Subscribe to project updates by watching https://github.com/bitnami/containers
mongodb 02:54:13.35 Submit issues and feature requests at https://github.com/bitnami/containers/issues
mongodb 02:54:13.35 
mongodb 02:54:13.36 INFO  ==> ** Starting MongoDB setup **
mongodb 02:54:13.37 INFO  ==> Validating settings in MONGODB_* env vars...
mongodb 02:54:13.47 INFO  ==> Initializing MongoDB...
sed: can't read /opt/bitnami/mongodb/conf/mongodb.conf: No such file or directory
```

**Comment by carrodher** [comment]
Are you able to reproduce the issue using the default values?

**Comment by Jaysins** [comment]
No, the default architecture is standalone. Which works. 

But changing it to replicaset causes this issue.

**Comment by Jaysins** [comment]
architecture: standalone

Changed to

architecture: replicaset

**Comment by Jaysins** [comment]
> In order to reproduce the issue on our side, could you please share the parameters that were modified?

have you been able to?

**Comment by joancafom** [comment]
Hi @Jaysins 

I can't reproduce the issue on my side:

```console
$ cat my-values.yaml
architecture: replicaset

$ helm install mongo -f my-values.yaml oci://registry-1.docker.io/bitnamicharts/mongodb
$ kubectl get pods
NAME                      READY   STATUS              RESTARTS   AGE
mongo-mongodb-0           1/1     Running             0          59s
mongo-mongodb-arbiter-0   1/1     Running             0          59s

$ kubectl logs mongo-mongodb-0
Warning: Use tokens from the TokenRequest API or manually created secret-based tokens instead of auto-generated secret-based tokens.
mongodb 10:39:42.39 INFO  ==> Advertised Hostname: mongo-mongodb-0.mongo-mongodb-headless.test.svc.cluster.local
mongodb 10:39:42.39 INFO  ==> Advertised Port: 27017
realpath: /bitnami/mongodb/data/db: No such file or directory
architecture: replicaset
mongodb 10:39:42.40 INFO  ==> Data dir empty, checking if the replica set already exists
MongoNetworkError: getaddrinfo ENOTFOUND mongo-mongodb-1.mongo-mongodb-headless.test.svc.cluster.local
mongodb 10:39:43.18 INFO  ==> Pod name matches initial primary pod name, configuring node as a primary
mongodb 10:39:43.19 INFO  ==>
mongodb 10:39:43.20 INFO  ==> Welcome to the Bitnami mongodb container
mongodb 10:39:43.20 INFO  ==> Subscribe to project updates by watching https://github.com/bitnami/containers
mongodb 10:39:43.20 INFO  ==> Submit issues and feature requests at https://github.com/bitnami/containers/issues
mongodb 10:39:43.20 INFO  ==>
mongodb 10:39:43.21 INFO  ==> ** Starting MongoDB setup **
mongodb 10:39:43.23 INFO  ==> Validating settings in MONGODB_* env vars...
mongodb 10:39:43.29 INFO  ==> Initializing MongoDB...
mongodb 10:39:43.35 INFO  ==> Writing keyfile for replica set authentication...
mongodb 10:39:43.36 INFO  ==> Deploying MongoDB from scratch...
MongoNetworkError: connect ECONNREFUSED 10.20.0.11:27017
mongodb 10:39:44.95 INFO  ==> Creating users...
mongodb 10:39:44.96 INFO  ==> Creating root user...
```

What kind of cluster are you deploying the asset in?

Sometimes, it is useful to enable the `debug` option in order to obtain more detailed logs. Could you please do that by specifying that in your installation params?

```diff
+image:
+  debug: true
architecture: replicaset
```

Could you please provide the output of the following commands as well?

- `kubectl get pvc`
- `kubectl describe pod/mongo-mongodb-0`

**Comment by Jaysins** [comment]
helm install mongo \
--set architecture=replicaset \
--set automountServiceAccountToken=false \
--set externalAccess.enabled=true \
--set externalAccess.autoDiscovery.enabled=true \
--set externalAccess.service.type=LoadBalancer \
--set externalAccess.service.port=27017 \
--set pdb.create=true \
--set resources.requests.memory="1Gi" \
--set resources.requests.cpu="500m" \
--set resources.limits.memory="2Gi" \
--set resources.limits.cpu="1" \
--set rbac.create=true \
--set replicaCount=3 \
--namespace mongo-staging-v7 \
--create-namespace \
oci://registry-1.docker.io/bitnamicharts/mongodb




Can you help pinpoint what's wrong with this setup? I have a v1.25 EKS cluster and trying to deploy a production ready mongo. I stopped using the .yaml files because I kept getting that error, but now, with --set, the pod runs, however running 

mongosh admin --host "mongo-mongodb-0.mongo-mongodb-headless.mongo-staging-v2.svc.cluster.local:27017,mongo-mongodb-1.mongo-mongodb-headless.mongo-staging-v2.svc.cluster.local:27017,mongo-mongodb-2.mongo-mongodb-headless.mongo-staging-v2.svc.cluster.local:27017" --authenticationDatabase admin -u root -p mypassword


returns with MongoServerSelectionError: Server selection timed out after 30000 ms


please help

**Comment by serkan1st1** [comment]
sed: can't read /opt/bitnami/mongodb/conf/mongodb.conf: No such file or directory 

I also get this error. Have you found a solution?

**Comment by Jaysins** [comment]
@serkan1st1  your yaml file should only contain parameters you want to set. Do not include the whole template.

just create a yaml fill with the conf necessary

architecture: replicaset 
replicaCount: 3 

if it still doesn't work use the --set

**Comment by Jaysins** [comment]
@joancafom 

This is my current setup to install mongo on helm

helm upgrade mongo \
--set architecture=replicaset \
--set replicaCount=3 \
--set persistence.size=500Gi \
--set automountServiceAccountToken=true \
--set externalAccess.enabled=true \
--set externalAccess.autoDiscovery.enabled=true \
--set externalAccess.service.type=LoadBalancer \
--set externalAccess.service.port=27017 \
--set resources.requests.memory="1Gi" \
--set resources.requests.cpu="1000m" \
--set resources.limits.memory="2Gi" \
--set resources.limits.cpu="1500m" \
--set rbac.create=true \
--set containerSecurityContext.runAsGroup=0 \
--set containerSecurityContext.runAsUser=0 \
--set rbac.create=true \
--set readinessProbe.initialDelaySeconds=240 \
--set arbiter.enabled=false \
--namespace mongo-staging-v2 \
--create-namespace \
oci://registry-1.docker.io/bitnamicharts/mongodb



however, my pod never leaves the not ready state and when i run

kubectl describe pod

  ----     ------     ----                 ----     -------
  Warning  Unhealthy  12s (x1224 over 3h)  kubelet  Readiness probe failed: Warning: Could not access file: ENOENT: no such file or directory, mkdir '/.mongodb/mongosh'
Error: Not ready

kubectl get all


NAME                  READY   STATUS    RESTARTS       AGE
pod/mongo-mongodb-0   0/1     Running   1 (3h3m ago)   3h4m

NAME                               TYPE           CLUSTER-IP       EXTERNAL-IP                                                              PORT(S)           AGE
service/mongo-mongodb-0-external   LoadBalancer   172.20.180.59    abf1bb010918b484cba59eb58e7c01a6-206515190.us-east-2.elb.amazonaws.com   27017:31696/TCP   3h4m
service/mongo-mongodb-1-external   LoadBalancer   172.20.184.248   a79280639dcd24dcabda0014bc300160-181666778.us-east-2.elb.amazonaws.com   27017:30144/TCP   3h4m
service/mongo-mongodb-2-external   LoadBalancer   172.20.169.79    a4606f81b135a42fca2fc070f375fc24-657973497.us-east-2.elb.amazonaws.com   27017:30759/TCP   3h4m
service/mongo-mongodb-headless     ClusterIP      None             <none>                                                                   27017/TCP         3h4m

NAME                             READY   AGE
statefulset.apps/mongo-mongodb   0/3     3h4m



and when I check logs

jayson@jayson-IdeaPad-5-15ITL05:~$ kubectl logs -f mongo-mongodb-0

Defaulted container "mongodb" out of: mongodb, auto-discovery (init)
mongodb 11:31:59.24 INFO  ==> Advertised Hostname: abf1bb010918b484cba59eb58e7c01a6-206515190.us-east-2.elb.amazonaws.com
mongodb 11:31:59.25 INFO  ==> Advertised Port: 27017
mongodb 11:31:59.25 INFO  ==> Pod name matches initial primary pod name, configuring node as a primary
mongodb 11:31:59.27 INFO  ==> 
mongodb 11:31:59.27 INFO  ==> Welcome to the Bitnami mongodb container
mongodb 11:31:59.27 INFO  ==> Subscribe to project updates by watching https://github.com/bitnami/containers
mongodb 11:31:59.27 INFO  ==> Submit issues and feature requests at https://github.com/bitnami/containers/issues
mongodb 11:31:59.28 INFO  ==> Upgrade to Tanzu Application Catalog for production environments to access custom-configured and pre-packaged software components. Gain enhanced features, including Software Bill of Materials (SBOM), CVE scan result reports, and VEX documents. To learn more, visit https://bitnami.com/enterprise
mongodb 11:31:59.28 INFO  ==> 
mongodb 11:31:59.28 INFO  ==> ** Starting MongoDB setup **
mongodb 11:31:59.30 INFO  ==> Validating settings in MONGODB_* env vars...
mongodb 11:31:59.34 INFO  ==> Initializing MongoDB...
mongodb 11:31:59.40 INFO  ==> Writing keyfile for replica set authentication...
mongodb 11:31:59.42 INFO  ==> Enabling authentication...
mongodb 11:31:59.42 INFO  ==> Deploying MongoDB with persisted data...
mongodb 11:31:59.46 INFO  ==> ** MongoDB setup finished! **

mongodb 11:31:59.50 INFO  ==> ** Starting MongoDB **
{"t":{"$date":"2024-03-25T11:31:59.555Z"},"s":"I",  "c":"CONTROL",  "id":5760901, "ctx":"main","msg":"Applied --setParameter options","attr":{"serverParameters":{"enableLocalhostAuthBypass":{"default":true,"value":false}}}}

**Comment by Jaysins** [comment]
Apparently to by pass this you first have to install the bitnami/mongo, without the externalservice enabled

helm install mongo \
--set architecture=replicaset \
--set replicaCount=3 \
--set persistence.size=500Gi \
--set resources.requests.memory="1Gi" \
--set resources.requests.cpu="1000m" \
--set resources.limits.memory="2Gi" \
--set resources.limits.cpu="1500m" \
--set readinessProbe.initialDelaySeconds=60 \
--set readinessProbe.periodSeconds=80 \
--set readinessProbe.timeoutSeconds=60 \
--set arbiter.enabled=false \
--namespace mongo-staging-v2 \
--create-namespace \
oci://registry-1.docker.io/bitnamicharts/mongodb



then upgrade, with the options present

helm upgrade mongo \
--set architecture=replicaset \
--set replicaCount=3 \
--set persistence.size=500Gi \
--set automountServiceAccountToken=true \
--set resources.requests.memory="1Gi" \
--set resources.requests.cpu="1000m" \
--set resources.limits.memory="2Gi" \
--set resources.limits.cpu="1500m" \
--set externalAccess.enabled=true \
--set automountServiceAccountToken=true \
--set externalAccess.enabled=true \
--set externalAccess.autoDiscovery.enabled=true \
--set externalAccess.service.type=LoadBalancer \
--set externalAccess.service.port=27017 \
--set rbac.create=true \
--set serviceAccount.create=true \
--set readinessProbe.initialDelaySeconds=60 \
--set readinessProbe.periodSeconds=80 \
--set readinessProbe.timeoutSeconds=60 \
--set arbiter.enabled=false \
--namespace mongo-staging-v2 \
oci://registry-1.docker.io/bitnamicharts/mongodb


however the external URL wont be accessible via mongosh

mongosh admin --host "ab75995378cc845aab64ad1660c36d32-360240056.us-east-2.elb.amazonaws.com:1111,ac13f8860e5514f29801b9c4b3228c90-1477742163.us-east-2.elb.amazonaws.com:30308,a8aeeb5b97a2445ed9f37b413f992078-1309354780.us-east-2.elb.amazonaws.com:31904" --authenticationDatabase admin -u username -p password

returns MongoNetworkError: getaddrinfo ENOTFOUND


mongo --host "ab75995378cc845aab64ad1660c36d32-360240056.us-east-2.elb.amazonaws.com:32246,ac13f8860e5514f29801b9c4b3228c90-1477742163.us-east-2.elb.amazonaws.com:30308,a8aeeb5b97a2445ed9f37b413f992078-1309354780.us-east-2.elb.amazonaws.com:31904" --authenticationDatabase admin -u username -p password

this works

**Comment by joancafom** [comment]
Hi @serkan1st1, I still can't hit the issue 😞... It would be nice if you shared more details of your set up. What kind of cluster are you using?

**Comment by joancafom** [comment]
Were you able to solve the issue related to the `mongodb.conf` file @Jaysins ? Maybe you can share some inputs in here as well.

Regarding your new concerns, let me just say that at first sight it seems to me you are using a wrong endpoint in your `mongosh` command

> XXXX.us-east-2.elb.amazonaws.com:1111

Beware that `1111` does not seem to be the correct port number.

At the same time, you refer to the same one correctly in your `mongo` command later:

> XXXX.us-east-2.elb.amazonaws.com:32246

Having said that, if you are now experiencing a new (different) issue, could you please raise a new ticket? We try to have single-themed issues so that users can easily find solutions to similar problems

**Comment by theKwabena** [comment]
Hi, @joancafom, I have the same issue with the mongodb.conf,
`sed: can't read /opt/bitnami/mongodb/conf/mongodb.conf: No such file or directory`
I think it has to do with the image tag. Try reproducing with image tag **4.4-debian-10**.
This is my values.yaml

architecture: replicaset
replicaCount: 3
image:
  registry: docker.io
  repository: bitnami/mongodb
  tag: 4.4-debian-10

**Comment by rrileyca** [comment]
We upgraded from 14.13.0 to 15.1.0. Same issues, ReplicaSet configuration won't leave the Ready state in Kubernetes. The healthcheck is failing.

When jump into the pod and exec `db.isMaster().isMaster` and `db.isMaster().secondary`, both are false. For some reason the new pod in the Stateful set won't join the other 2 that exists. I can confirm DNS and network connectivity is working.

Enabling debug doesn't seem helpful, as the logs look like the other running pods in the StatefulSet:
```text
Defaulted container "mongodb" out of: mongodb, metrics, generate-tls-certs (init), auto-discovery (init)
mongodb 18:18:51.02 INFO  ==> Advertised Hostname: <IP Address omitted>
mongodb 18:18:51.02 INFO  ==> Advertised Port: 27017
mongodb 18:18:51.03 INFO  ==> Pod name doesn't match initial primary pod name, configuring node as a secondary
mongodb 18:18:51.04 INFO  ==>
mongodb 18:18:51.04 INFO  ==> Welcome to the Bitnami mongodb container
mongodb 18:18:51.04 INFO  ==> Subscribe to project updates by watching https://github.com/bitnami/containers
mongodb 18:18:51.05 INFO  ==> Submit issues and feature requests at https://github.com/bitnami/containers/issues
mongodb 18:18:51.05 INFO  ==> Upgrade to Tanzu Application Catalog for production environments to access custom-configured and pre-packaged software components. Gain enhanced features, including Software Bill of Materials (SBOM), CVE scan result reports, and VEX documents. To learn more, visit https://bitnami.com/enterprise
mongodb 18:18:51.05 INFO  ==>
mongodb 18:18:51.05 DEBUG ==> Copying files from /opt/bitnami/mongodb/conf.default to /opt/bitnami/mongodb/conf
mongodb 18:18:51.06 INFO  ==> ** Starting MongoDB setup **
mongodb 18:18:51.07 INFO  ==> Validating settings in MONGODB_* env vars...
mongodb 18:18:51.12 INFO  ==> Initializing MongoDB...
mongodb 18:18:51.13 DEBUG ==> /mongodb.conf mounted. Skipping setting port and IPv6 settings
mongodb 18:18:51.13 DEBUG ==> /mongodb.conf mounted. Skipping setting log settings
mongodb 18:18:51.13 DEBUG ==> /mongodb.conf mounted. Skipping setting log settings
mongodb 18:18:51.13 DEBUG ==> /mongodb.conf mounted. Skipping setting storage settings
mongodb 18:18:51.14 INFO  ==> Writing keyfile for replica set authentication...
mongodb 18:18:51.14 DEBUG ==> /mongodb.conf mounted. Skipping keyfile location configuration
mongodb 18:18:51.15 DEBUG ==> /mongodb.conf mounted. Skipping authorization enabling
mongodb 18:18:51.15 INFO  ==> Deploying MongoDB with persisted data...
mongodb 18:18:51.15 DEBUG ==> /mongodb.conf mounted. Skipping replicaset mode enabling
mongodb 18:18:51.15 DEBUG ==> /mongodb.conf mounted. Skipping authorization enabling
mongodb 18:18:51.16 DEBUG ==> /mongodb.conf mounted. Skipping IP binding to all addresses
mongodb 18:18:51.16 DEBUG ==> Skipping loading custom scripts on non-primary nodes...

mongodb 18:18:51.16 INFO  ==> ** MongoDB setup finished! **
mongodb 18:18:51.18 INFO  ==> ** Starting MongoDB **
{"t":{"$date":"2024-04-04T18:18:51.224Z"},"s":"I",  "c":"CONTROL",  "id":5760901, "ctx":"main","msg":"Applied --setParameter options","attr":{"serverParameters":{"enableLocalhostAuthBypass":{"default":true,"value":false}}}}
```

One thing I've noticed is the root filesystem, most things are owned by `root` but now mongo is running as UID `1001`. Is it possible this patch broke upgrades and not new installations?

**Comment by nowost** [comment]
Hello!  Have you found a solution? I get this error too.

**Comment by kladiv** [comment]
During my first upgrade i got the same issue
- Helm chart upgrade from 14.8.0 to 15.1.3
- Image tag: 5.0.24 (image ref was `5.0.24-debian-11-r2`).

> Using command `crictl inspecti docker.io/bitnami/mongodb:5.0.24` i found
> org.opencontainers.image.ref.name": "5.0.24-debian-11-r2"

Some values:
```
    architecture: standalone
    useStatefulSet: true
    tls:
      enabled: true
      mode: allowTLS
      certificatesSecret: mongodb-tls-selfsigned
      certFilename: tls.crt
      certKeyFilename: tls.key
    extraFlags:
      - --tlsAllowConnectionsWithoutCertificates
```

I found it does not download the latest image ref of 5.0.24 so i forced the image upgrade
- Image tag: 5.0.24 (image ref is now `5.0.24-debian-11-r20`)

So try to force upgrade to latest digest of your Image version (or add `image.pullPolicy: Always` in helm values)

**Comment by joancafom** [comment]
Hi everyone, thanks for the input!

I can reproduce the issue if I follow the steps in recent replies (in which the chart is being upgraded from version 14.x.x to 15.x.x but still uses an old image). First of all, let me explain why this happens... 

We recently released the new **major** version `15.x.x` of the `bitnami/mongodb` chart, you can see the release notes [here](https://github.com/bitnami/charts/tree/main/bitnami/mongodb#to-1500). In this new major, the security defaults have been updated and the chart runs with more secure options by default. 

These changes required some previous adaptation in the docker images, that were included in time in them before the major release of the chart. **This means that older versions/tags (that do not include those changes) of the `bitnami/mongo` image will not work properly out-of-the-box with the new version of the chart**.

If you still want to use older images in the chart, you can revert the new security defaults as explained [in the release notes](https://github.com/bitnami/charts/tree/main/bitnami/mongodb#to-1500).

**Comment by kladiv** [comment]
Hi @joancafom,
i'm using your latest MongoDB 5.0 image ([5.0.24-debian-11-r20](https://hub.docker.com/layers/bitnami/mongodb/5.0.24-debian-11-r20/images/sha256-e56a75744316419cd150400ccd8d985c6b0762f03c7a3b015f233524d043731f?context=explore)) but we still got some issue

If MongoDB 5.0 (and 6.0) images are not supported in Helm chart version 15.x, probabily it should be better to add in the Upgrade release notes.

> In MongoDB 7.0 latest images the file indicated in the error appears to have the correct permissions instead:
> ```
> I have no name!@6c78d2c024b1:/opt/bitnami/mongodb$ ls -ld /opt/bitnami/mongodb/conf.default/mongodb.conf
> -rw-r--r-- 1 root root 1018 Apr  9 20:26 /opt/bitnami/mongodb/conf.default/mongodb.conf
> ```

If possible, MongoDB 5.0 and 6.0 (which are still not EOL) should be updated to be compatible to 15.x helm chart version

**Comment by nowost** [comment]
i found solution, need specify
```
  configuration: |
    # mongod.conf
    .......
```

**Comment by etolstanov** [comment]
I had the same problem while trying to apply some new values to my chart.

In my case, I'm running a sharded cluster on Kubernetes, but fixed it by upgrading the image.

Non-working image:   `bitnami/mongodb-sharded:6.0.11-debian-11-r1`
Working image: `bitnami/mongodb-sharded:6.0.13-debian-11-r20`

Hope this helps somebody, I almost lost my entire cluster this morning 😢

**Comment by rrileyca** [comment]
So the question then is - when will Bitnami use the functional images in the Helm Chart?

**Comment by juan131** [comment]
Hi everyone,

Is anyone facing issues using latest MongoDB chart version (**15.1.4**) and the latest image included by default in the chart (**7.0.8-debian-12-r2**)?

> If possible, latest MongoDB 5.0 and 6.0 images (which are still not EOL) should be fixed to be compatible to 15.x.y helm chart version

@kladiv is not possible for Bitnami to overwrite a previously published image. The workaround to use a previos version that isn't compatible with the new security defaults is to revert these defaults as explained [in the release notes](https://github.com/bitnami/charts/tree/main/bitnami/mongodb#to-1500).

**Comment by kladiv** [comment]
> @kladiv is not possible for Bitnami to overwrite a previously published image. The workaround to use a previos version that isn't compatible with the new security defaults is to revert these defaults as explained [in the release notes](https://github.com/bitnami/charts/tree/main/bitnami/mongodb#to-1500).

They could release newer versions of the images like 5.0.24-debian-11-**r21** or 6.0.13-debian-11-**r22** (not present at the moment)

**Comment by juan131** [comment]
Hi @kladiv @lombardialess 

I'm afraid we cannot release new versions of the `5.x` and `6.x` branches since MongoDB is not publishing binaries for Debian 12.

It's not also possible for our systems to release revisions of versions that we no longer support. Therefore, for those using these versions, I'm afraid the only alternative is to revert the security defaults as I previously mentioned.

**Comment by juan131** [comment]
Hi @kladiv 

We have some internal discussions currently about creating changelog(s) in the charts so we can document this kind of stuff. More news to come.

Regarding "option b", we do support `5.x` and `6.x` in the [VMware Tanzu Application Catalog](https://bitnami.com/enterprise), the enterprise edition of Bitnami Application Catalog which support other distros (e.g. Ubuntu, RedHat or PhotonOS). However, it's not possible to support them for Debian 12 (the one used on Bitnami Application Catalog) since MongoDB  is not publishing binaries for this distro.

**Comment by tgfarrell** [comment]
My concern is there is a serious security CVE for MongoDB ( https://jira.mongodb.org/browse/SERVER-72839 ) being addressed in version 6.0.14 and we can't get the binary on the https://hub.docker.com/r/bitnami/mongodb tagged as 6.0.14. 

BTW, the tags are showing as debian 11 not 12. 

Did the chart change to using debian 12 for the 6.x MongoDB container?

**Comment by juan131** [comment]
Hi @tgfarrell 

> My concern is there is a serious security CVE for MongoDB ( https://jira.mongodb.org/browse/SERVER-72839 ) being addressed in version 6.0.14 and we can't get the binary on the https://hub.docker.com/r/bitnami/mongodb tagged as 6.0.14.

Unfortunately, no new releases will be provided in the `6.x` branch in the Bitnami Application Catalog. Switching the distro version is sth that we can exclusively do at catalog level (affecting every single Bitnami image) so we cannot make any more Debian 11 release for MongoDB.

> BTW, the tags are showing as debian 11 not 12.
> Did the chart change to using debian 12 for the 6.x MongoDB container?

The chart changed to use, by default, MongoDB `7.x` at https://github.com/bitnami/charts/commit/2587a913f1f8ad325870d94673f87ecaf1781098, then Debian 11 images were replace by Debian 12 images at https://github.com/bitnami/charts/commit/96c398f7cca8f220ce707e2d15349cbcc74de9af

**Comment by tgfarrell** [comment]
> Hi @tgfarrell
> 
> > My concern is there is a serious security CVE for MongoDB ( https://jira.mongodb.org/browse/SERVER-72839 ) being addressed in version 6.0.14 and we can't get the binary on the https://hub.docker.com/r/bitnami/mongodb tagged as 6.0.14.
> 
> Unfortunately, no new releases will be provided in the `6.x` branch in the Bitnami Application Catalog. Switching the distro version is sth that we can exclusively do at catalog level (affecting every single Bitnami image) so we cannot make any more Debian 11 release for MongoDB.
> 
> > BTW, the tags are showing as debian 11 not 12.
> > Did the chart change to using debian 12 for the 6.x MongoDB container?
> 
> The chart changed to use, by default, MongoDB `7.x` at [2587a91](https://github.com/bitnami/charts/commit/2587a913f1f8ad325870d94673f87ecaf1781098), then Debian 11 images were replace by Debian 12 images at [96c398f](https://github.com/bitnami/charts/commit/96c398f7cca8f220ce707e2d15349cbcc74de9af)

So no more support going forward for 5.x ot 6.x versions of MongoDB with the Bitnami charts? Thats unfortunate. I may have to build my own Debian 11 MongoDB image then for our chart.
